#338 Group messaging

Pagebakers opened the pull request:
Fixes #338 #488

ShishKabab commented:
Some initial thoughts looking at the code without yet understanding the big picture how everything is connected:
- components/chat/conversation-settings.jsx and maybe other files, a few instances of potentially unbound event handlers
- src/js/components/chat/conversation.jsx, does getParticipant() method belong in that component?
- Profile is being passed down as context, but should either come from store or props
- '// omit current user' appears twice, factor out into different functions? In general, I'd keep preproccessing data outside of components. This also includes getting display names for example
- In general: one or max. two classes per file would be nice
- ContactsList: could some of the logic be factored out?
- ChatAgent is quite big. Could some responsibilities be factored out? Are there method that don't need to manipulate the state directly, and can be separated out, or made static?

These are some things I've noticed, but don't know if they're bugs or missing features:
- No notification when receiving new message
- It doesn't update on the receiving end (no polling observed.)

Pagebakers commented:
Some initial thoughts looking at the code without yet understanding the big picture how everything is connected:
- components/chat/conversation-settings.jsx and maybe other files, a few instances of potentially unbound event handlers
Fixed

- src/js/components/chat/conversation.jsx, does getParticipant() method belong in that component?
Agreed, should take this in consideration with the Redux migration.

- Profile is being passed down as context, but should either come from store or props
Out of scope for this issue.

- '// omit current user' appears twice, factor out into different functions? In general, I'd keep preproccessing data outside of components. This also includes getting display names for example
Optimize in Redux migration

- In general: one or max. two classes per file would be nice
Agreed & fixed, I got lazy here

- ContactsList: could some of the logic be factored out?
I was thinking about that too, out of scope for now.

- ChatAgent is quite big. Could some responsibilities be factored out? Are there method that don't need to manipulate the object state directly, and can be separated out, or made static? Maybe we should seperate out nearly stateless classes talking to the backend, from classes that do have a more sophisticated internal state?
I'm open for suggestions, but out of scope of this issue.

These are some things I've noticed, but don't know if they're bugs or missing features:
- No notification when receiving new message
Notifications haven't been build in. (should be handled on the server trough push notifications)

- It doesn't update on the receiving end (no polling observed.)
Fixed, I accidentally didn't uncomment the polling line.
